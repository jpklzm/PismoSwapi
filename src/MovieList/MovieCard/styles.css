.movie {
  background-color: rgb(100, 100, 100, 0.5);
  padding: 15px 30px;
  border-radius: 10px;
  margin: 5px;
  color: yellow;
  cursor: pointer;
}

.movie:hover {
  background-color: rgb(100, 100, 100, 1);
}
