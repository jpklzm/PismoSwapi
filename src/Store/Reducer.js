const Reducer = (state, action) => {
  switch (action.type) {
    case "SET_CURRENT_MOVIE":
      return {
        ...state,
        movie: action.payload,
      };
    case "SET_CURRENT_MOVIE_LIST":
      return {
        ...state,
        movieList: action.payload,
      };
    default:
      return state;
  }
};

export default Reducer;
