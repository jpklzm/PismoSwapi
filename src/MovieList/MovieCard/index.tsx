import React, { FunctionComponent, useContext, useCallback } from "react";
import "./styles.css";
import { Context } from "../../Store/Store";
import { useHistory } from "react-router-dom";

type MovieCardProps = {
  movie: { title: string };
};

const MovieCard: FunctionComponent<MovieCardProps> = ({ movie }) => {
  //@ts-ignore
  const [state, dispatch] = useContext(Context);

  const history = useHistory();
  const handleOnClick = useCallback(
    (currentMovie) => {
      dispatch({ type: "SET_CURRENT_MOVIE", payload: currentMovie });

      history.push("/movie-details");
    },
    [history]
  );

  return (
    <div>
      {movie && (
        <>
          <div
            className="movie"
            onClick={() => {
              handleOnClick(movie);
            }}
          >
            <h2>{movie.title}</h2>
          </div>
        </>
      )}
    </div>
  );
};

export default MovieCard;
