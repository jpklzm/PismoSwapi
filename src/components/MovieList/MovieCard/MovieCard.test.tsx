import { mount } from "enzyme";
import MovieCard from "./index";
import { Context } from "../../Store/Store";

const contextValues = {
  movieList: [{ title: "filme" }, { title: "outrofilme" }],
};

describe("MovieCard", () => {
  it("renders correctly", () => {
    const wrapper = mount(
      <Context.Provider value={[contextValues, ""]}>
        <MovieCard />
      </Context.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
