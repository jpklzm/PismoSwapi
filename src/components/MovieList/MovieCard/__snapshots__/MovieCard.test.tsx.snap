// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders correctly 1`] = `
<MovieCard>
  <div />
</MovieCard>
`;
